build-tools/packages/build-tools/src/type-test-generator/main.ts

```diff
@@ -62,25 +63,42 @@ const currentFile = new Project({
 
 const previousTsConfigPath = `${previousBasePath}/tsconfig.json`;
 let previousFile: SourceFile;
-if (existsSync(previousTsConfigPath)) {
-	const project = new Project({
-		skipFileDependencyResolution: true,
-		tsConfigFilePath: previousTsConfigPath,
-	});
+const project = new Project({
+	skipFileDependencyResolution: true,
+	tsConfigFilePath: existsSync(previousTsConfigPath) ? previousTsConfigPath : undefined,
+});
+// Check for existence of alpha and add appropriate file
+if (existsSync(alphaFilePath)) {
+	project.addSourceFilesAtPaths(alphaFilePath);
+	previousFile = project.getSourceFileOrThrow("<package-name>-alpha.d.ts");
+	// If alpha doesn't exist but index.ts and tsconfig.json do
+} else if (existsSync(previousTsConfigPath)) {
 	previousFile = project.getSourceFileOrThrow("index.ts");
+	// Fall back to using .d.ts
 } else {
-	const project = new Project({
-		skipFileDependencyResolution: true,
-	});
 	project.addSourceFilesAtPaths(`${previousBasePath}/dist/**/*.d.ts`);
 	previousFile = project.getSourceFileOrThrow("index.d.ts");
```

Comment:
this should actually be where we check for alpha. not in the current. types not in the previous but in the current will be ignored by default.